spec/factories/courses.rb

```diff
@@ -22,7 +22,9 @@
 
     trait :with_video_component_enabled do
       after(:build) do |course|
-        course.instance.set_component_enabled_boolean(:course_videos_component, true)
+        # Save instance setting to the database so the video component remains enabled on reload.
+        # Should help stop seemingly random video component spec failures.
+        course.instance.set_component_enabled_boolean!(:course_videos_component, true)
```

Comment:
Instead of forcing a save here would it be better to fix the specs to save the course instead? 

It then makes sense to add some documentation to say that this trait doesn't necessarily work unless course is persisted.

Reply:
I think it would be better to have the factory work as expected instead of having to remember that courses using this trait have to be saved?

Reply:
Hmm, I think my suggestion was because there might be a case where I might only need to build but not persist the couse object, so the introduction of this code unintentionally saves my course instead. 

Your point is valid, but only if it doesn't introduce too much behaviour that might be hard to customise.

Reply:
As discussed, having the factory not persist the object allows different behaviour to be injected for testing or be used in a more flexible way as a stub.

However, this is currently outweighed by having to track this down as the cause for any other new spec which uses this trait.

The comment added helps to explain the reasoning behind adding the save to the factory.